pystring.cpp

```diff
@@ -1200,15 +1209,15 @@ namespace path
                 //     4. join('c:', 'd:/') = 'd:/'
                 //     5. join('c:/', 'd:/') = 'd:/'
                 
-                if( (pystring::slice(path, 1, 2) != ":") ||
-                    (pystring::slice(b, 1, 2) == ":") )
+                if( (pystring::slice(path, 1, 2) != colon) ||
```

Comment:
Same here. Creating new strings just to find out if one character is a colon seems wasteful.

Reply:
same here.